apps/browser/src/platform/services/abstractions/script-injector.service.ts

```diff
@@ -0,0 +1,28 @@
+export type Mv2ScriptInjectionDetails = {
+  file: string;
+};
+
+export type Mv3ScriptInjectionDetails = {
+  file: string;
+  world?: chrome.scripting.ExecutionWorld; // Optional, defaults to "ISOLATED"
+};
+
+/**
+ * Configuration for injecting a script into a tab. The `file` property should present as a
+ * path that is relative to the root directory of the extension build, ie "content/script.js".
+ */
+export type ScriptInjectionConfig = {
+  tabId: number;
+  injectDetails: {
+    file?: string; // Should only be set if the script should be called in both manifest versions. If the mv2Details or mv3Details are set, the file will be pulled from those details.
+    allFrames?: boolean; // Optional, defaults to false. If set to true, the script will injection into all frames in the tab
+    frameId?: number; // Optional, defaults to 0, which is the id of the top level frame in a tab
+    runAt?: "document_start" | "document_end" | "document_idle"; // Optional, defaults to "document_start"
+  };
```

Comment:
⛏️ JSDoc format lets these values be seen by the editor.

🎨 Since the parameters are all `?`, I've removed "optional" from your descriptions. Also included @coroiu's grammar change and a link to the MDN docs explaining the `runAt` parameter.

```suggestion
    /** Script injected into the document. 
     *  Overridden by `mv2Details` and `mv3Details`. 
     */
    file?: string;
    /** When `true`, the script injects into every frame in the tab. Defaults to `false`. */
    allFrames?: boolean;
    /** Identifies the frame targeted for script injection. Defaults to the top level frame (0). */
    frameId?: number; 
    /** When the script executes. Defaults to "document_start".
     * @see https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/content_scripts 
     */
    runAt?: "document_start" | "document_end" | "document_idle";
  };
```

Reply:
Sounds good, I've applied these and updated the `world` mv3 reference to have a link to documentation in the same manner as your `run_at` suggestion.